etcd-client/src/lib.rs

```diff
@@ -219,67 +225,153 @@ mod tests {
+                    .await
+                    .unwrap();
+                client
+                    .kv()
+                    .put(EtcdPutRequest::new(key2.as_str(), "baz3"))
+                    .await
+                    .unwrap();
+                smol::Timer::after(Duration::from_secs(1)).await;
 
-        let watch_key = "41_foo1";
-        let mut response = client.watch(KeyRange::key(watch_key)).await;
-        assert_eq!(
-            response
-                .next()
-                .await
-                .unwrap_or_else(|| panic!("Fail to receive reponse from client"))
-                .unwrap_or_else(|e| { panic!("failed to get watch response, the error is: {}", e) })
-                .watch_id(),
-            0,
-            "Receive wrong watch response from etcd server"
-        );
-
-        client
-            .kv()
-            .put(EtcdPutRequest::new("41_foo1", "baz3"))
-            .await?;
-        let mut watch_response = response
-            .next()
-            .await
-            .unwrap_or_else(|| panic!("Fail to receive reponse from client"))
-            .unwrap_or_else(|e| panic!("failed to get watch response, the error is: {}", e));
+                log::debug!("delete");
+                client
+                    .kv()
+                    .delete(EtcdDeleteRequest::new(KeyRange::key(key1.as_str())))
+                    .await
+                    .unwrap();
+                client
+                    .kv()
+                    .delete(EtcdDeleteRequest::new(KeyRange::key(key2.as_str())))
+                    .await
+                    .unwrap();
+            })
+            .detach();
+        }
 
-        assert_eq!(
-            watch_response.watch_id(),
-            0,
-            "Receive wrong watch response from etcd server"
-        );
-        assert_eq!(
-            watch_response.take_events().len(),
-            1,
-            "Receive wrong watch events from etcd server"
-        );
+        let join = {
+            let client = client.clone();
+            let watchkey = key1.clone();
+            smol::spawn(async move {
+                watch_one(watchkey.as_str(), client).await;
+            })
+        };
+        let join2 = {
+            let client = client.clone();
+            let watchkey = key1.clone();
+            smol::spawn(async move {
+                watch_one(watchkey.as_str(), client).await;
+            })
+        };
+        let join3 = {
+            let client = client.clone();
+            let watchkey = key2.clone();
+            smol::spawn(async move {
+                watch_one(watchkey.as_str(), client).await;
+            })
+        };
+        let join4 = {
+            let client = client.clone();
+            let watchkey = key2.clone();
+            smol::spawn(async move {
+                watch_one(watchkey.as_str(), client).await;
+            })
+        };
+        join.await;
+        join2.await;
+        join3.await;
+        join4.await;
```

Comment:
https://docs.rs/futures/0.3.28/futures/future/fn.join_all.html. Try this api.

Reply:
Ok.